metricbeat/module/elasticsearch/elasticsearch.go

```diff
@@ -359,6 +360,45 @@ func GetXPack(http *helper.HTTP, resetURI string) (XPack, error) {
 	return xpack, err
 }
 
+type IndexSettings struct {
+	Hidden bool
+}
+
+// GetIndicesSettings returns a map of index names to their settings.
+// Note that as of now it is optimized to fetch only the "hidden" index setting to keep the memory
+// footprint of this function call as low as possible.
+func GetIndicesSettings(http *helper.HTTP, resetURI string) (map[string]IndexSettings, error) {
+	content, err := fetchPath(http, resetURI, "*/_settings", "filter_path=*.settings.index.hidden&expand_wildcards=all")
+
+	if err != nil {
+		return nil, errors.Wrap(err, "could not fetch indices settings")
+	}
+
+	var resp map[string]struct {
+		Settings struct {
+			Index struct {
+				Hidden string `json:"hidden"`
+			} `json:"index"`
```

Comment:
why not `Hidden bool`, or `Index IndexSettings json:"index"`?

Reply:
It had to do with the type of `index.hidden` in the Elasticsearch API's JSON response being a string, e.g. `"true"`, instead of being a JSON boolean, e.g. `true`. Let me make a commit showing what I tried before this implementing `json.Unmarshaler` then we can decide which implementation we like better.

Reply:
@urso I've made an alternate implementation (the one I tried originally) in 87b73dda8. Let me know if you prefer that one right before that. Thanks!

Reply:
Uff, I see. If we would use something like `boolstr` more often we might consider to provide a package/library with decoding primitives. But here as a standalone single use case within the package I think I prefer to decode into string first. A hybrid solution  could be `Index struct { Hidden boolstr `json:"index"` }`, but keep `IndexSetting{ Hidden bool }`. Then you might be able to cast from `Index` into `IndexSetting` directly. Using `boolstr` we see decoding errors directly returned from the JSON decoder if something changes.

Reply:
Thanks, let me try the hybrid solution next.

Reply:
@urso Implemented the hybrid approach in eb621b5ce825a622aae008506d1ee3507be44c94. Let me know what you think. Thanks!